[wip] Don't use codegen'ed boxing wrappers in c10 for factory ops

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31967 [wip] Don't use codegen'ed boxing wrappers in c10 for factory ops**
* #30370 [wip] Use codegen'ed unboxing wrappers

Differential Revision: [D19320662](https://our.internmc.facebook.com/intern/diff/D19320662/)

smessmer committed:
[wip] Don't use codegen'ed boxing wrappers in c10 for factory ops



Differential Revision: [D19320662](https://our.internmc.facebook.com/intern/diff/D19320662/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 4406f4cb:


* **1/2** failures introduced in this PR
* **1/2** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=4406f4cb6a689564978cb6969e6427ceafc7301c).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4177060) caffe2_onnx_py2_gcc5_ubuntu16_04_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=82288749) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=82288749)) :snowflake:
<details>
<summary>
<code>Jan 08 23:17:19 unknown file: Failure</code>
</summary>

```
Jan 08 23:17:19 frame #10: /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test() [0x449515] 
Jan 08 23:17:19 frame #11: /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test() [0x44976d] 
Jan 08 23:17:19 frame #12: testing::internal::UnitTestImpl::RunAllTests() + 0xbc2 (0x44a732 in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test) 
Jan 08 23:17:19 frame #13: testing::UnitTest::Run() + 0x8b (0x44aa4b in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test) 
Jan 08 23:17:19 frame #14: main + 0x3c (0x417a5c in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test) 
Jan 08 23:17:19 frame #15: __libc_start_main + 0xf0 (0x7f1065260830 in /lib/x86_64-linux-gnu/libc.so.6) 
Jan 08 23:17:19 frame #16: _start + 0x29 (0x417e09 in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test) 
Jan 08 23:17:19 " thrown in the test body. 
Jan 08 23:17:19 [  FAILED  ] BackendFallbackTest.TestBackendFallbackWithMode (7 ms) 
Jan 08 23:17:19 [ RUN      ] BackendFallbackTest.TestBackendFallbackWithWrapper 
Jan 08 23:17:19 unknown file: Failure 
Jan 08 23:17:19 C++ exception with description "false INTERNAL ASSERT FAILED at /var/lib/jenkins/workspace/aten/src/ATen/core/boxing/KernelFunction_impl.h:46, please report a bug to PyTorch. Tried to call KernelFunction::callBoxed() on a KernelFunction that can only be called with KernelFunction::callUnboxed(). (callBoxed at /var/lib/jenkins/workspace/aten/src/ATen/core/boxing/KernelFunction_impl.h:46) 
Jan 08 23:17:19 frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x6a (0x7f1065e5650a in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/lib/libc10.so) 
Jan 08 23:17:19 frame #1: c10::KernelFunction::callBoxed(c10::OperatorHandle const&, std::vector<c10::IValue, std::allocator<c10::IValue> >*) const + 0x559 (0x7f10667c6ea9 in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/lib/libtorch_cpu.so) 
Jan 08 23:17:19 frame #2: <unknown function> + 0x2d35437 (0x7f1068d9f437 in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/lib/libtorch_cpu.so) 
Jan 08 23:17:19 frame #3: /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test() [0x41ab20] 
Jan 08 23:17:19 frame #4: /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test() [0x41bc17] 
t::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, bool, double, double, bool>(void (*)(c10::OperatorKernel*, c10::OperatorHandle const&, std::vector<c10::IValue, std::allocator<c10::IValue> >*), c10::OperatorKernel*, c10::OperatorHandle const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, bool, double, double, bool) + 0x176 (0x428166 in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test) 
Jan 08 23:17:19 frame #6: /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test() [0x41b8db] 
Jan 08 23:17:19 frame #7: /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test() [0x41d46e] 
Jan 08 23:17:19 frame #8: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) + 0x43 (0x453463 in /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/test/backend_fallback_test) 
```

</details>

### 1 failure *not* recognized by patterns:
| Job | Step | Status |
| --- | --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>caffe2_onnx_py3_6_clang7_ubuntu16_04_test</sup> | <sup>Test</sup> | New in PR |


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "[wip] Don't use codegen'ed boxing wrappers in c10 for factory ops"


Differential Revision: [D19320662](https://our.internmc.facebook.com/intern/diff/D19320662/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Don't use codegen'ed boxing wrappers in c10 for factory ops"


Differential Revision: [D19320662](https://our.internmc.facebook.com/intern/diff/D19320662/)

[ghstack-poisoned]
